### PR TITLE
Retry registry/buildkit transient errors

### PR DESCRIPTION
**Pull request by pchico83**
Signed-off-by: Pablo Chico de Guzman <pchico83@gmail.com>

### CODE REVIEW

**Review by jLopezbarb — APPROVED** [review]

File: pkg/cmd/build/build.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 	"os"
 	"path/filepath"
 
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
```
Comment (rberrelleza) [inline pkg/cmd/build/build.go:21]: we use `okErrors` in other parts of the repo. 

```suggestion
	okErrors "github.com/okteto/okteto/pkg/errors"
```

File: pkg/errors/errors.go [hunk 2]
```diff
@@ -131,6 +131,21 @@ func IsTransient(err error) bool {
 	}
 }
 
+// IsRegistryTransient returns true if err represents a transient registry error
+func IsRegistryTransient(err error) bool {
```
Comment (rberrelleza) [inline pkg/errors/errors.go:135]: can we instead put this in the `registry`  package, so the call is something like `registry.IsTransientError(err error)` or something like that? that way we keep the registry-related business logic in a single package.

**Review by rberrelleza — APPROVED** [review]

File: pkg/cmd/build/build.go [hunk 3]
```diff
@@ -59,5 +60,10 @@ func Run(ctx context.Context, namespace, buildKitHost string, isOktetoCluster bo
 		return errors.Wrap(err, "failed to create build solver")
 	}
 
-	return solveBuild(ctx, buildkitClient, opt, progress)
+	err = solveBuild(ctx, buildkitClient, opt, progress)
+	if oktetoErrors.IsRegistryTransient(err) {
+		log.Yellow("Retrying registry transient error...")
```
Comment (rberrelleza) [inline pkg/cmd/build/build.go:65]:
```suggestion
		log.Yellow("Failed to push  %s to the registry,  retrying ...",  image)
```

File: pkg/errors/errors.go [hunk 4]
```diff
@@ -131,6 +131,21 @@ func IsTransient(err error) bool {
...
+	if err == nil {
+		return false
+	}
+
+	switch {
+	case strings.Contains(err.Error(), "failed commit on ref") && strings.Contains(err.Error(), "500 Internal Server Error"),
```
Comment (rberrelleza) [inline pkg/errors/errors.go:141]: do you need all three messages to be in the error, or are we handling three different errors here?

File: pkg/cmd/build/build.go (hunk 3 above)
Comment (rberrelleza) [reply pkg/cmd/build/build.go:65]: should we track this event?

**Review by rberrelleza — APPROVED** [review]

File: pkg/errors/errors.go (hunk 4 above)
Comment (pchico83) [reply pkg/errors/errors.go:141]: they are two different errors, one of them with two conditions

File: pkg/registry/registry.go [hunk 5]
```diff
@@ -110,3 +110,18 @@ func ExpandOktetoDevRegistry(ctx context.Context, namespace, tag string) (string
...
 }
+
+// IsTransientError returns true if err represents a transient registry error
+func IsTransientError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	switch {
+	case strings.Contains(err.Error(), "failed commit on ref") && strings.Contains(err.Error(), "500 Internal Server Error"),
```
Comment (rberrelleza) [inline pkg/registry/registry.go:121]: This is not very easy to grasp, can you split into two different `case` statements so it's easier to see the conditions?

**Pull request merged**